src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -46,6 +47,28 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses {@code oneBasedIndex} into an {@code Index} and returns it. Leading and trailing whitespaces will be
+     * trimmed. This is specifically used for DeleteCommand.
+     * @throws ParseException if the specified index is invalid (less than 1 or not an integer).
+     */
+    public static Index parseDeleteIndex(String oneBasedIndex) throws ParseException {
+        String trimmedIndex = oneBasedIndex.trim();
+        if (trimmedIndex.isEmpty()) {
+            throw new ParseException("");
```

Comment:
Is there a better way to throw this error upwards? Don't think empty error message is good practice. Maybe change to a more relevant message?

Reply:
Fixed in latest commit

Reply:
@wj331 Need to fix checkstyle otherwise lgtm